File: Frontend/apps/astrogame_app/lib/models/technologies/player_dependent_finished_technology.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/technologies/finished_technology.dart';
import 'package:flutter_guid/flutter_guid.dart';
import 'package:json_annotation/json_annotation.dart';

part 'player_dependent_finished_technology.g.dart';

@GuidConverter()
@JsonSerializable()
class PlayerDependentFinishedTechnology extends FinishedTechnology {
  @JsonKey()
  Guid playerId;

  PlayerDependentFinishedTechnology();

  factory PlayerDependentFinishedTechnology.fromJson(
          Map<String, dynamic> json) =>
      _$PlayerDependentFinishedTechnologyFromJson(json);
  Map<String, dynamic> toJson() =>
      _$PlayerDependentFinishedTechnologyToJson(this);
}

File: Backend/AstroGame.Shared/Models/Technologies/PlayerDependentFinishedTechnology.cs
using System;
using AstroGame.Shared.Models.Players;

namespace AstroGame.Shared.Models.Technologies
{
    public class PlayerDependentFinishedTechnology : FinishedTechnology
    {
        public Guid PlayerId { get; set; }

        public virtual Player Player { get; set; }
    }
}